From 21a939418e823838713e49213211c1221cc8a657 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 116/296] MountService: catch IllegalStateException for
 getField

An IllegalStateException will be thrown when getFiled is timeout.
So catch it to avoid crash.
---
 services/core/java/com/android/server/StorageManagerService.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/services/core/java/com/android/server/StorageManagerService.java b/services/core/java/com/android/server/StorageManagerService.java
index 8c93f372e09..902302c6d3e 100644
--- a/services/core/java/com/android/server/StorageManagerService.java
+++ b/services/core/java/com/android/server/StorageManagerService.java
@@ -1028,6 +1028,8 @@ class StorageManagerService extends IStorageManager.Stub
         String systemLocale;
         try {
             systemLocale = getField(StorageManager.SYSTEM_LOCALE_KEY);
+        } catch (IllegalStateException e) {
+            return;
         } catch (RemoteException e) {
             return;
         }
-- 
2.11.0

